tests/queries/0_stateless/02763_last_day_of_week.sql

```diff
@@ -0,0 +1,17 @@
+WITH 
+    toDate('2018-12-25') + number AS d, 
+    toDate32(d) AS d32, 
+    toDateTime(d) AS dt, 
+    toDateTime64(d, 0) AS dt64
+SELECT
+    dt64,
+    toLastDayOfWeek(d) AS wd_0,
+    toLastDayOfWeek(d32) AS wd32_0,
+    toLastDayOfWeek(dt) AS wdt_0,
+    toLastDayOfWeek(dt64) AS wdt64_0,
+    toLastDayOfWeek(d, 3) AS wd_3,
```

Comment:
Just to double-check: Out of the [different modes](https://clickhouse.com/docs/en/sql-reference/functions/date-time-functions#toweek) in `toWeek`, it only makes sense to distinguish cases "week starts on Monday" vs. "week starts on Sunday", right?

(maybe we could use case "3" here instead of "1" then)?

Reply:
I didn't quite get it I'm afraid. I agree that there is only one significant difference here - Monday vs Sunday, so we could juxtapose here any of the couples - for example, 0 and 1.
Maybe you meant that we could use case "1" here instead of "3"?

Reply:
Yes, I meant "1" instead of "3". Sorry, I mixed it up.

Reply:
Got it, thanks.

Reply:
Done. Now this test case resides in the existing `00941_to_custom_week` test and uses more straightforward `mode` argument (`1` instead of `3`). And I feel a good deal of temptation to similarly refine the existing test case regarding the `toStartOfWeek` function - something like this:

![image](https://github.com/ClickHouse/ClickHouse/assets/29695184/d76255c9-1965-4d8f-b747-e069d14580e7)

However, I seldom dare to walk along this precarious trail and to modify the existing code without a permission or direct recommendation.

Reply:
Thanks.

About modifying existing code (and that's just my opinion): Not doing it (i.e. only ever adding new code) means that things become more inconsistent over time and eventually deteriorate into a mess. I have seen that in software which shall not be named, so it's better to not let this happen in the first place. But I am not particularly pedantic about it, of course, and mixing changes with too many related cleanup can be a mess in itself.

Reply:
I couldn’t agree with you more. So, I pushed the commit that aligns the old test case of `toStartOfWeek` function with new extended test case of `toLastDayOfWeek` function.